.circleci/config.yml

```diff
@@ -92,385 +108,894 @@ commands:
+            gcloud --quiet config set project $GOOGLE_PROJECT_ID
+            gcloud --quiet config set compute/zone $GOOGLE_COMPUTE_ZONE
+            gcloud auth configure-docker --quiet << pipeline.parameters.container_registry >>
+          environment:
+            GKE_CLUSTER_NAME: <<parameters.cluster >>
+
+  create_gke_cluster:
+    parameters:
+      cluster:
+        description: "cluster name"
+        type: string
+        default: << pipeline.parameters.gcp_cluster_name >>
+      no-output-timeout:
+        description: >
+          Elapsed time that the cluster creation command can run on CircleCI without
+          output. The string is a decimal with unit suffix, such as “20m”, “1.25h”,
+          “5s”
+        type: string
+        default: 15m
+      num_nodes:
+        description: "Number of nodes to create"
+        type: integer
+        default: 1
+      machine_type:
+        description: "GKE cluster machine type"
+        type: string
+        default: "n1-standard-4"
+      disk_size:
+        description: "GKE cluster disk size"
+        type: integer
+        default: 100
+      disk_type:
+        description: "GKE cluster disk type"
+        type: string
+        default: "pd-ssd"
+      gpu_type:
+        description: "GKE cluster GPU type"
+        type: string
+        default: "nvidia-tesla-t4"
+      gpu_count:
+        description: "GKE cluster GPU count"
+        type: integer
+        default: 1
+    steps:
+      - setup_gcloud
+      - run:
+          name: "Check if cluster exists and delete if it does"
+          command: |
+            cluster_exists=$(gcloud container clusters list --filter="name=<< parameters.cluster >>" --format="value(name)" | wc -l | xargs)
+            gcloud container clusters list --filter="name=<< parameters.cluster >>" --format="value(name)"
+            if [ $? -eq 0 ] && [ $cluster_exists -eq 1 ]; then
+              gcloud container clusters delete << parameters.cluster >> --quiet
+            fi
+            exit 0
+      - run:
+          name: "Create GKE cluster"
+          command: |
+            gcloud container clusters create $GKE_CLUSTER_NAME \
+            --num-nodes=<< parameters.num_nodes >> --machine-type=<< parameters.machine_type >> \
+            --disk-size=<< parameters.disk_size >> --disk-type=<< parameters.disk_type >> \
+            --accelerator=type=<< parameters.gpu_type >>,count=<< parameters.gpu_count >>
+#            list gpus available in zone:
+#            gcloud compute accelerator-types list | grep us-central1
+#            TPU support
+#           gcloud beta container clusters update cluster-name --enable-tpu
+#           gcloud container clusters get-credentials
+#            TPU support
+#            --release-channel=stable \
+#            --scopes=cloud-platform \
+#            --enable-ip-alias \
+#            --enable-tpu
+#            e2-medium
+#            --accelerator=type=nvidia-tesla-k80,count=1
```

Comment:
commented out code? for future changes?

Reply:
yea, i can take this out, it's just "notes on the margins" for the upcoming tpu pod.

Reply:
ah that's fine... just keep it :)